c2rust-transpile/src/convert_type.rs

```diff
@@ -134,14 +132,14 @@ pub const RESERVED_NAMES: [&str; 103] = [
 ];
 
 impl TypeConverter {
-    pub fn new(emit_no_std: bool) -> TypeConverter {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> TypeConverter {
         TypeConverter {
             translate_valist: false,
             renamer: Renamer::new(&RESERVED_NAMES),
             fields: HashMap::new(),
             suffix_names: HashMap::new(),
             features: HashSet::new(),
-            emit_no_std,
         }
```

Comment:
Don't suppress the warning; follow it.  We should `impl Default for TypeConverter`.  It might need an `impl Default for Renamer` that calls `Self::new(&RESERVED_NAMES)`.

Reply:
Even with `emit_no_std` as a field, it still could have been `#[derive(Default)]`ed if there is an

```rust
impl Default for Renamer {
    fn default() -> Self {
        Self::new(&RESERVED_NAMES);
    }
}
```
which seems like a good Rust default (and the only time it's used outside of tests).  Then `new` could be (before this PR):

```rust
pub fn new(emit_no_std: bool) -> Self {
    Self {
        emit_no_std,
        ..Default::default()
    }
}
```

and (after this PR):

```rust
pub fn new() -> Self {
    Self::default()
}
```

`TypeConverter` seems like a type that will always have defaults, and so I think it should `impl Default`, regardless of if `new` takes arguments or not.  That said, the point about future compatibility and re-adding parameters is a good point, though I think any future parameters/fields would have a default, like `emit_no_std` did.  And if we do go this route, you should include that above explanation in a comment above `#[allow(clippy::new_without_default)]`.

Reply:
Ok, I've added an explanation.

Reply:
I still think it should be `#[derive(Default)]`ed.